lib/optimize/RemoveParentModulesPlugin.js

```diff
@@ -58,12 +61,17 @@ class RemoveParentModulesPlugin {
 	apply(compiler) {
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin(["optimize-chunks-basic", "optimize-extracted-chunks-basic"], (chunks) => {
-				chunks.forEach((chunk) => {
-					const cache = {};
-					chunk.modules.slice().forEach((module) => {
-						if(chunk.parents.length === 0) return;
-						const dId = "$" + debugIds(module.chunks);
-						let parentChunksWithModule;
+				for(var index = 0, lenChunks = chunks.length; index < lenChunks; index++) {
+					var chunk = chunks[index];
+
+					var cache = {};
```

Comment:
That is a very sad result for `Map` 😢 

Could you add a TODO at these locations `TODO consider Map when performance has improved`? Maybe with a link to the gist

I played a bit with your benchmark and noticed that node 6 and node 7 have different performance depending on the cache entry is cached:

https://gist.github.com/sokra/b36098368da7b8f6792fd7c85fca6311

``` text
// node.js v6.9.1
Map x 435 ops/sec ±2.23% (80 runs sampled)
Native x 1,697 ops/sec ±1.30% (84 runs sampled)
NativeSingleAccess x 8,176 ops/sec ±1.09% (86 runs sampled)
Fastest is NativeSingleAccess
```

``` text
// node.js v7.4.0
Map x 432 ops/sec ±4.84% (76 runs sampled)
MapSingleAccess x 804 ops/sec ±3.61% (82 runs sampled)
Native x 7,046 ops/sec ±3.14% (83 runs sampled)
NativeSingleAccess x 7,010 ops/sec ±2.86% (80 runs sampled)
Fastest is Native,NativeSingleAccess
```

Should we optimize our code for performance on LTS or stay with latest?